### PR TITLE
opt out convergence by default

### PR DESCRIPTION
**Pull request by manishtomar**
All tenants will be convergence enabled unless opted out via `"non-convergence-tenants"` config. "convergence-tenants" key is not used anymore. One has to put `none` in above key to enable convergence in all tenants instead of `all` earlier.

NOTE: This will be merged and deployed after getting a go-ahead from product. Requires https://github.com/rackerlabs/autoscaling-chef/pull/806 to be merged before this.

### CODE REVIEW

File: otter/convergence/composition.py [hunk 1]
```diff
@@ -22,12 +22,12 @@ def tenant_is_enabled(tenant_id, get_config_value):
         "convergence-tenants" portion of the configuration file.
     :param callable get_config_value: config key -> config value.
     """
-    enabled_tenant_ids = get_config_value("convergence-tenants")
```
Comment (radix) [inline otter/convergence/composition.py]: I recommend making this an additive change so we can switch from a whitelist to a blacklist with only a config change.

Here's a suggestion for how: `convergence-enabled` could be a new config key, the value would either be `"blacklist"` or `"whitelist"`, and then there would also be `convergence-tenants` and `non-convergence-tenants` keys.

```python
if convergence_enabled == 'blacklist' and tenant_id not in non_convergence_tenants:
    return True
elif convergence_enabled == 'whitelist' and tenant_id in convergence_tenants:
    return True
else:
    return False
```

Comment (manishtomar) [inline otter/convergence/composition.py]: We can always have additive deploys by first deploying "non-convergence-tenants" config change followed by this PR change. Would those steps be comfortable?

Comment (radix) [inline otter/convergence/composition.py]: yeah, it makes sense to put off merging this until we want to make the change.

**Comment by radix** [comment]
This is looking good, thanks for the change manish :)

**Comment by manishtomar** [comment]
jenkins please run integration tests

**Comment by manishtomar** [comment]
jenkins please run integration tests

**Comment by manishtomar** [comment]
jenkins please run integration tests

**Pull request merged**